src/EntryPoints/WikibaseFacetedSearchHooks.php

```diff
@@ -224,4 +241,146 @@ public static function onCirrusSearchAddQueryFeatures( SearchConfig $config, arr
 		$extraFeatures[] = new HasWbFacetFeature();
 	}
 
+	/**
+	 * @param SearchIndexField[] $fields
+	 */
+	public static function onSearchIndexFields( array &$fields, SearchEngine $engine ): void {
+		if ( !( $engine instanceof CirrusSearch ) ) {
+			return;
+		}
+
+		$fields = array_merge(
+			$fields,
+			WikibaseFacetedSearchExtension::getInstance()->newFacetSearchIndexFieldsBuilder( $engine )->createFields()
+		);
+	}
+
+	public static function onSearchDataForIndex(
+		array &$fields,
+		ContentHandler $handler,
+		WikiPage $page,
+		ParserOutput $output,
+		SearchEngine $engine
+	): void {
+		$config = WikibaseFacetedSearchExtension::getInstance()->getConfig();
+
+		if ( $config->linkTargetSitelinkSiteId === null ) {
+			$entityId = WikibaseRepo::getEntityIdLookup()->getEntityIdForTitle( $page->getTitle() );
+
+			if ( $entityId !== null ) {
+				$fields = array_merge( $fields, self::getFieldValues( $entityId ) );
+			}
+
+			return;
+		}
+
+		if ( !( $handler instanceof WikitextContentHandler ) ) {
+			return;
+		}
+
+		try {
+			$entityId = WikibaseRepo::getStore()->newSiteLinkStore()->getEntityIdForLinkedTitle(
+				$config->linkTargetSitelinkSiteId,
+				$page->getTitle()->getText()
+			);
+		} catch ( Exception $exception ) {
+			return;
+		}
+
+//		print_r("\nentityId:$entityId");
+
+		if ( $entityId === null ) {
+			return;
+		}
+
+		$fields = array_merge( $fields, self::getFieldValues( $entityId ) );
+	}
+
+	private static function getFieldValues( EntityId $entityId ): array {
+		$entity = WikibaseRepo::getEntityLookup()->getEntity( $entityId );
```

Comment:
I suspect something like this will be nice:

```php
private function getStatements(): StatementList {
		return $this->getStatementsLookup()->getStatements();
	}
	
	private function getStatementsLookup(): StatementsLookup {
		if ( /* sitelink config case */ ) {
			return new SitelinkBasedStatementLookup( /* sitelink config */ );
		}
		
		return new FromPageStatementLookup( $wikiPage /* or similar */ );
	}
```

To encapsulate the logic for getting the statements for both the cases we support (wiki with sitelink config or not).

I switched from Entity to StatementList in the signatures, since we only need the later for our current stuff. Could argue for keeping Entity to potentially support aliases and similar data that one might want facets for, but then again YAGNI.